openeo-geotrellis/src/test/scala/org/openeo/geotrellis/MergeCubesSpec.scala

```diff
@@ -121,6 +144,63 @@ class MergeCubesSpec {
     assertTrue(mse < 0.1)
   }
 
+
+  @ParameterizedTest
+  @MethodSource(Array("testMergeCubesTiledNodataArguments"))
+  def testMergeCubesTiledNodata(aggregateR: AggregationType.Value,
+                                aggregateG: AggregationType.Value,
+                                aggregateB: AggregationType.Value,
+                               ): Unit = {
+    val path = "tmp/testMergeCubesTiledNodata/" + aggregateR + aggregateG + aggregateB + "/"
+    Files.createDirectories(Paths.get(path))
+    val p = new OpenEOProcesses()
+
+    def agg(rdd: ContextRDD[SpaceTimeKey, MultibandTile, TileLayerMetadata[SpaceTimeKey]],
+            aggregationType: AggregationType.Value,
+           ): ContextRDD[SpaceTimeKey, MultibandTile, TileLayerMetadata[SpaceTimeKey]] = {
+      val startDate = rdd.keys.collect().head.time
+      if (aggregationType == AggregationType.no) {
+        rdd
+      } else {
+        val intervals = if (aggregationType == AggregationType.extraNoData) {
+          List(startDate, startDate, startDate.plusMonths(1), startDate.plusMonths(1)).map(DateTimeFormatter.ISO_INSTANT.format(_)).asJava
+        } else {
+          List(startDate, startDate).map(DateTimeFormatter.ISO_INSTANT.format(_)).asJava
+        }
+
+        val labels = if (aggregationType == AggregationType.extraNoData) {
+          List(startDate, startDate.plusMonths(1)).map(DateTimeFormatter.ISO_INSTANT.format(_)).asJava
+        } else {
+          List(startDate).map(DateTimeFormatter.ISO_INSTANT.format(_)).asJava
+        }
+
+        val composite = p.aggregateTemporal(
+          rdd,
+          intervals,
+          labels,
+          TestOpenEOProcessScriptBuilder.createMedian(true),
+          java.util.Collections.emptyMap()
+        )
+        // composite = p.filterEmptyTile(composite)
+        val tmp2 = new ContextRDD(composite, composite.metadata)
+        tmp2
+      }
+    }
```

Comment:
I would have opted for an OO approach of defining a base trait with a single method (~ `agg`) and three implementing classes/objects to replace these enum values that don't have behavior.

Reply:
Sounds like more code, no? Ok if I leave it like this?